lib/Sema/TypeCheckAttr.cpp

```diff
@@ -874,13 +874,15 @@ void AttributeChecker::visitAccessControlAttr(AccessControlAttr *attr) {
       }
     } else {
       if (auto VD = dyn_cast<ValueDecl>(D)) {
-        // Emit warning when trying to declare non-@objc `open` member inside an
-        // extension.
-        if (!VD->isObjC() && attr->getAccess() == AccessLevel::Open) {
-          diagnose(attr->getLocation(),
-                   diag::access_control_non_objc_open_member,
-                   isa<AbstractFunctionDecl>(VD))
-              .fixItReplace(attr->getRange(), "public");
+        if (isa<AbstractFunctionDecl>(VD) || isa<AbstractStorageDecl>(VD)) {
```

Comment:
What about subscripts, initializers, etc.?

Reply:
Wouldn't they be included under `AbstractFunctionDecl` and `AbstractStorageDecl`? I was assuming we need to issue a warning for those cases as well; am I mistaken?

Reply:
Sorry, that’s my question to you, as I’m not sure; could it read more clearly explicitly to _exclude_ type declarations?

Reply:
I added test cases for subscripts, initializers and nominal types. Subscripts will work as open if they're @objc, but initializers won't work as initializers can't be open anyways. 
Also, I changed the expression to exclude `NominalTypeDecl` instead :)